com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/HooksHelper.java

```diff
@@ -0,0 +1,163 @@
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.util.MtaMetadataUtil;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaApplication;
+import com.sap.cloud.lm.sl.cf.core.model.HookPhase;
+import com.sap.cloud.lm.sl.cf.process.steps.ProcessContext;
+import com.sap.cloud.lm.sl.cf.process.steps.StepPhase;
+import com.sap.cloud.lm.sl.cf.process.steps.StepsUtil;
+import com.sap.cloud.lm.sl.cf.process.variables.Variables;
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.Hook;
+import com.sap.cloud.lm.sl.mta.model.Module;
+
+public class HooksHelper {
+
+    private final MtaMetadataParser mtaMetadataParser;
+    private final EnvMtaMetadataParser envMtaMetadataParser;
+    private final ProcessContext context;
+    private final HookPhase hookPhaseBeforeStep;
+    private final HookPhase hookPhaseAfterStep;
+
+    private HooksHelper(Builder hooksHelperBuilder) {
+        this.mtaMetadataParser = hooksHelperBuilder.mtaMetadataParser;
+        this.envMtaMetadataParser = hooksHelperBuilder.envMtaMetadataParser;
+        this.context = hooksHelperBuilder.context;
+        this.hookPhaseBeforeStep = hooksHelperBuilder.hookPhaseBeforeStep;
+        this.hookPhaseAfterStep = hooksHelperBuilder.hookPhaseAfterStep;
+    }
+
+    public List<Hook> getHooksForStepPhase(Module moduleToDeploy, StepPhase currentStepPhase) {
+        HookPhase currentHookPhaseForExecution = determineHookPhaseForCurrentStepPhase(currentStepPhase);
+        return getHooksForCurrentPhase(moduleToDeploy, currentHookPhaseForExecution);
+    }
+
+    public Module determineModuleToDeploy(ProcessContext context) {
+        Module moduleToDeploy = context.getVariable(Variables.MODULE_TO_DEPLOY);
+        return moduleToDeploy != null ? moduleToDeploy : determineModuleFromDescriptor(context);
+    }
+
+    public void setHooksForExecution(List<Hook> hooksForExecution) {
+        context.setVariable(Variables.HOOKS_FOR_EXECUTION, hooksForExecution);
+    }
+
+    private Module determineModuleFromDescriptor(ProcessContext context) {
+        DeploymentDescriptor deploymentDescriptor = context.getVariable(Variables.COMPLETE_DEPLOYMENT_DESCRIPTOR);
+        if (deploymentDescriptor == null) {
+            // This will be the case only when the process is undeploy.
+            return null;
+        }
+
+        CloudApplicationExtended cloudApplication = context.getVariable(Variables.APP_TO_PROCESS);
+        if (cloudApplication.getModuleName() == null) {
+            // This case handles the deletion of old applications when the process is blue-green deployment. Here the application is taken
+            // from the
+            // CloudController and thus we do not have moduleName in it.
+            return determineModuleFromAppName(deploymentDescriptor, cloudApplication);
+        }
+
+        HandlerFactory handlerFactory = StepsUtil.getHandlerFactory(context.getExecution());
```

Comment:
We can use this in the `determineModuleFromAppName` method as well, so move it above the second if condition.
Also, since that `determine` method does the same as the `findModuleByNameFromDeploymentDescriptor` method only with one more processing of the module name, we can refactor this method so as to "determine" the module name that needs checking in the if statement (whether it's directly from the cloud application object, or after mta metadata / env parsing) and then passing it to the `findModule` method.
Some renaming may be worthwhile and maybe some extracting into methods